order- or main-related compilation error ?

**V version:** 
0.1.17
**OS:** 
MacOS 10.14.6
**What did you do?**
tried to compile the following:
println('hello')

struct User {
    id int 
    name string
} 

struct Repo {
    users []User 
}

**What did you expect to see?**
successful compilation, because it does compile if I put the println() underneath the structs
 
**What did you see instead?**
get_type() bad type
pass=2 fn=`main`
panic: break.v:9
unknown type `User`

After reading some more documentation I was confused about what to expect:
* should the compiler have warned about a missing main()?
* is this supposed to work in the first place?

--- a/compiler/parser.v
+++ b/compiler/parser.v
@@ -403,21 +403,18 @@ fn (p mut Parser) parse(pass Pass) {
 			if p.is_script && !p.pref.is_test {
 				// cur_fn is empty since there was no fn main declared
 				// we need to set it to save and find variables
-				if p.first_pass() {
-					if p.cur_fn.name == '' {
-						p.set_current_fn( MainFn )
-					}
-					return
-				}
 				if p.cur_fn.name == '' {
 					p.set_current_fn( MainFn )
 					if p.pref.is_repl {
+						if p.first_pass() {
+							return
+						}
 						p.clear_vars()
 					}
 				}
 				mut start := p.cgen.lines.len
 				p.statement(true)
-				if p.cgen.lines[start - 1] != '' && p.cgen.fn_main != '' {
+				if start > 0 && p.cgen.lines[start - 1] != '' && p.cgen.fn_main != '' {
 					start--
 				}
 				p.genln('')

